features/gui/admin_panel/functions.lua

```diff
@@ -0,0 +1,160 @@
+local Report = require 'features.report'
+
+-- == ACTIONS =================================================================
+
+local Actions = require 'features.admin_commands'
+
+---@param filename? string
+---@param player? LuaPlayer
+function Actions.save_game(filename, player)
+  filename = filename or 'currently-running'
+  game.auto_save(filename)
+  local msg = 'Saving map: _autosave-' .. filename ..'.zip'
+  if player and player.valid then
+    player.print(msg)
+  else
+    game.print(msg)
+  end
+end
+
+---@param player? LuaPlayer
+function Actions.delete_all_blueprints(player)
+  local count = 0
+  for _, surface in pairs(game.surfaces) do
+    for _, ghost in pairs(surface.find_entities_filtered { type = { 'entity-ghost', 'tile-ghost' }}) do
+      ghost.destroy()
+      count = count + 1
+    end
+  end
+  local msg = count .. ' blueprints removed from all surfaces.'
+  if player and player.valid then
+    player.print(msg)
+  else
+    game.print(msg)
+  end
+end
+
+---@param player? LuaPlayer
+function Actions.destroy_all_speakers(player)
+  local count = 0
+  for _, surface in pairs(game.surfaces) do
+    for _, speaker in pairs(surface.find_entities_filtered { type = 'programmable-speaker' }) do
+      if speaker.parameters.playback_globally == true then
+        speaker.die('player')
+        count = count + 1
+      end
+    end
+  end
+  local msg = count .. ' speakers removed from all surfaces.'
+  if player and player.valid then
+    player.print(msg)
+  else
+    game.print(msg)
+  end
+end
+
+---@param player? LuaPlayer
+function Actions.kill_all_units(player)
```

Comment:
```suggestion
function Actions.kill_all_enemy_units(player)
```
The button captions are fine.